src/agent/srcview/src/srcview.rs

```diff
@@ -49,6 +49,42 @@ impl SrcView {
         Ok(self.0.insert(module.to_owned(), cache))
     }
 
+    /// Insert a new pdb into the SrcView only if the `pdb` path is not in the SrcView already,
+    /// returning a  [Result] indicating the success of the insert, if any was necessary.
+    /// If an insert was not necessary, returns [Ok].
+    /// 
+    /// # Arguments
+    /// 
+    /// * `module` - Module name to store the PDB info as
+    /// * `pdb` - Path to PDB
+    /// 
+    /// # Errors
+    /// 
+    /// If the PDB at the provided path has not been inserted already **and**
+    /// cannot be parsed or contains otherwise unexpected data.
+    /// 
+    /// # Example
+    /// 
+    /// ```no_run
+    /// use srcview::SrcView;
+    /// 
+    /// let mut sv = SrcView::new();
+    /// let modoffs = get_coverage();
+    /// 
+    /// // Map each modoff to a PDB name/path and make sure it's in the SrcView
+    /// for modoff in modoffs {
+    ///     let module_name =  mod_name_from_modoff(modoff);
+    ///     sv.try_insert(module_name, format!("~/pdbs/{module_name}.pdb"));
+    /// }
+    /// ```
+    pub fn try_insert<P: AsRef<Path>>(&mut self, module: &str, pdb: P) -> Result<()> {
```

Comment:
It's often useful for the caller to know if an insertion happened or not, suggest returning `Result<bool>`?

Reply:
Great suggestion. Fixed.